tf_files/configs/kube-setup-fence.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+#
+# Deploy fence into existing commons - assume configs are already configured
+# for fence to re-use the userapi db.
+# This fragment is pasted into kube-services.sh by kube.tf.
+#
+
+set -e
+
+vpc_name=${vpc_name:-$1}
+if [ -z "${vpc_name}" ]; then
+   echo "Usage: bash kube-setup-fence.sh vpc_name"
+   exit 1
+fi
+if [ ! -d ~/"${vpc_name}" ]; then
+  echo "~/${vpc_name} does not exist"
+  exit 1
+fi
+
+cd ~/${vpc_name}_output
+python render_creds.py secrets
+
+cd ~/${vpc_name}
+# Generate RSA private and public keys.
+# TODO: generalize to list of key names?
+mkdir -p jwt-keys
+
+if [ ! -f jwt-keys/jwt_public_key.pem ]; then
+  openssl genrsa -out jwt-keys/jwt_private_key.pem 2048
+  openssl rsa -in jwt-keys/jwt_private_key.pem -pubout -out jwt-keys/jwt_public_key.pem
+fi
+if ! kubectl get configmaps/fence > /dev/null 2>&1; then
+  kubectl create configmap fence --from-file=apis_configs/user.yaml
+fi
+
+if ! kubectl get secrets/fence-secret > /dev/null 2>&1; then
+  kubectl create secret generic fence-secret --from-file=local_settings.py=./apis_configs/fence_settings.py
+fi
+
+if ! kubectl get secrets/fence-jwt-keys > /dev/null 2>&1; then
+  kubectl create secret generic fence-jwt-keys --from-file=./jwt-keys
+fi
+
+kubectl apply -f services/fence/fence-deploy.yaml
+
+if [[ -z "${fence_snapshot}" && "${create_fence_db}" = "true" && ( ! -f .rendered_fence_db ) ]]; then
```

Comment:
I didn't see `create_fence_db`templated by terraform?

Reply:
it's hard-coded to 'true' in kube-services.sh (via kube-services-body.sh):
    create_fence_db="true"

The assumption is that you'll run 'kube-services.sh' to setup a brand new VPC
otherwise - if you're running kube-setup-fence.sh directly, then the assumption
is that you're running kube-setup-fence.sh directly, so you're probably updating
and existing environment that already has a userapi db that fence will use.

You can always do:
   create_fence_db=true bash kube-setup-fence.sh vpcname

Note sure how to link to the changes: kube.tf :

     provisioner "local-exec" {
-        command = "echo \"${data.template_file.kube_vars.rendered}\" | cat - \"${path.module}/../configs/kube-certs.sh\" \"${path.module}/../configs/kube-services-body.sh\" > ${var.vpc_name}_output/kube-services.sh"
+        command = "echo \"${data.template_file.kube_vars.rendered}\" | cat - \"${path.module}/../configs/kube-setup-certs.sh\" \"${path.module}/../configs/kube-services-body.sh\" \"${path.module}/../configs/kube-setup-fence.sh\" > ${var.vpc_name}_output/kube-services.sh"
     }

kube-services-body.sh:

+#
+# Set a flag for the kube-setup-fence fragment
+# that terraform appends to kube-services.sh
+#
+create_fence_db="true"
+